### PR TITLE
add tests for client path parameters

### PR DESCRIPTION
**Pull request by iscai-msft**
# Cadl Ranch Contribution Checklist:

- [ ] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [ ] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/models/property-types/main.cadl [hunk 1]
```diff
@@ -172,4 +172,4 @@ interface DictionaryString extends ModelOperations<DictionaryStringProperty, "{'
 model NeverProperty is ModelTemplate<never>;
 @route("/never")
 @operationGroup
-interface Never extends ModelOperations<NeverProperty, "<dont include this property>"> {}
+interface Never extends ModelOperations<NeverProperty, "<don't include this property>"> {}
```
Comment (iscai-msft) [inline packages/cadl-ranch-specs/http/models/property-types/main.cadl:175]: fixed bc it failed cspell

File: packages/cadl-ranch-specs/http/clients/parameters/path/main.cadl [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+@scenario
+@scenarioDoc("""
+Expected response body:
+  ```json
+  {"id": 1}
+  ```
+""")
+@doc("Get a resource")
+op get is ResourceRead<Product>;
```
Comment (timotheeguerin) [inline packages/cadl-ranch-specs/http/clients/parameters/path/main.cadl:56]: I would just remove the resource part no? If we are testing server params maybe just 
```
op noOperationParams(): NoContentResponse;

// And maybe adding this one to test it can still add path param for operations
op withOperationPathParam(@path id: string): NoContentResponse;
```

like we have for authentication scenarios

Comment (iscai-msft) [reply packages/cadl-ranch-specs/http/clients/parameters/path/main.cadl:56]: I can split this up more, I kept it with templating bc we have api version both in the path, and the template will return api version as aquery param (this is what's happening in anomaly detector). Maybe the two operations you've listed, and then another one with a template to test the duplicate api versions?

File: packages/cadl-ranch-specs/http/models/property-types/main.cadl (hunk 1 above)
Comment (timotheeguerin) [reply packages/cadl-ranch-specs/http/models/property-types/main.cadl:175]: or did I merge without the ci passing...

**Comment by tadelesh** [comment]
@iscai-msft could you help to review?

**Review by timotheeguerin — APPROVED** [review]

**Comment by iscai-msft** [comment]
@tadelesh talked with @timotheeguerin , we don't feel the need to have a core test right now (in the future all core tests should also be in the azure core folder). We're then combining the existing parameterized test with the test you added with api version as a path aparameter and they'll both be under the `Server.Path` route. Thanks!
